In [4]:
import pandas as pd
import numpy as np

## 1. Загрузите файл HR.csv в pandas dataframe

In [5]:
hr = pd.read_csv('HR.csv')
hr.head(10)

satisfaction_level  last_evaluation  number_project  average_montly_hours  \
0                0.38             0.53               2                   157   
1                0.80             0.86               5                   262   
2                0.11             0.88               7                   272   
3                0.72             0.87               5                   223   
4                0.37             0.52               2                   159   
5                0.41             0.50               2                   153   
6                0.10             0.77               6                   247   
7                0.92             0.85               5                   259   
8                0.89             1.00               5                   224   
9                0.42             0.53               2                   142   

   time_spend_company  Work_accident  left  promotion_last_5years department  \
0                   3              0     1                      0      sales   
1                   6              0     1                      0      sales   
2                   4              0     1                      0      sales   
3                   5              0     1                      0      sales   
4                   3              0     1                      0      sales   
5                   3              0     1                      0      sales   
6                   4              0     1                      0      sales   
7                   5              0     1                      0      sales   
8                   5              0     1                      0      sales   
9                   3              0     1                      0      sales   

   salary  
0     low  
1  medium  
2  medium  
3     low  
4     low  
5     low  
6     low  
7     low  
8     low  
9     low

## 2. Рассчитайте основные статистики для переменных.


In [6]:
hr.describe()

satisfaction_level  last_evaluation  number_project  \
count        14999.000000     14999.000000    14999.000000   
mean             0.612834         0.716102        3.803054   
std              0.248631         0.171169        1.232592   
min              0.090000         0.360000        2.000000   
25%              0.440000         0.560000        3.000000   
50%              0.640000         0.720000        4.000000   
75%              0.820000         0.870000        5.000000   
max              1.000000         1.000000        7.000000   

       average_montly_hours  time_spend_company  Work_accident          left  \
count          14999.000000        14999.000000   14999.000000  14999.000000   
mean             201.050337            3.498233       0.144610      0.238083   
std               49.943099            1.460136       0.351719      0.425924   
min               96.000000            2.000000       0.000000      0.000000   
25%              156.000000            3.000000       0.000000      0.000000   
50%              200.000000            3.000000       0.000000      0.000000   
75%              245.000000            4.000000       0.000000      0.000000   
max              310.000000           10.000000       1.000000      1.000000   

       promotion_last_5years  
count           14999.000000  
mean                0.021268  
std                 0.144281  
min                 0.000000  
25%                 0.000000  
50%                 0.000000  
75%                 0.000000  
max                 1.000000

## 3. Рассчитайте и визуализировать корреляционную матрицу для количественных переменных.

### Определите две самые скоррелированные и две наименее скоррелированные переменные.


## 4. Рассчитайте сколько сотрудников работает в каждом департаменте.


In [7]:
len(hr['department'].unique())

10

In [8]:
hr['department'].value_counts().head(10)

sales          4140
technical      2720
support        2229
IT             1227
product_mng     902
marketing       858
RandD           787
accounting      767
hr              739
management      630
Name: department, dtype: int64

## 5. Показать распределение сотрудников по зарплатам. 

In [9]:
len(hr['salary'].unique())

3

In [10]:
r = hr['salary'].value_counts()
r.head()

low       7316
medium    6446
high      1237
Name: salary, dtype: int64

#### Вывод: сотрудники с высокой зарполатой - 1237 чел., сотрудники со средней зарплатой - 6446 чел., сотрудники с низкой зарплатой - 7316 чел.

## 6. Показать распределение сотрудников по зарплатам в каждом департаменте по отдельности


In [11]:
hr.groupby('department').salary.value_counts()

department   salary
IT           low        609
             medium     535
             high        83
RandD        medium     372
             low        364
             high        51
accounting   low        358
             medium     335
             high        74
hr           medium     359
             low        335
             high        45
management   high       225
             medium     225
             low        180
marketing    low        402
             medium     376
             high        80
product_mng  low        451
             medium     383
             high        68
sales        low       2099
             medium    1772
             high       269
support      low       1146
             medium     942
             high       141
technical    low       1372
             medium    1147
             high       201
Name: salary, dtype: int64

## 7. Проверить гипотезу, что сотрудники с высоким окладом проводят на работе больше времени, чем сотрудники с низким окладом


In [12]:
hr_ = hr[['salary','average_montly_hours']]
hr_.head()

salary  average_montly_hours
0     low                   157
1  medium                   262
2  medium                   272
3     low                   223
4     low                   159

In [13]:
m = hr_.groupby('salary').average_montly_hours.sum()
m.head()

salary
high       247236
low       1470491
medium    1297827
Name: average_montly_hours, dtype: int64

In [14]:
m / r

high      199.867421
low       200.996583
medium    201.338349
dtype: float64

#### Вывод: сотрудники со средним окладом проводят суммарно времени больше на рабочем мечте (но не на много)

# 8. Рассчитать следующие показатели среди уволившихся и неуволившихся сотрудников (по отдельности):

● Доля сотрудников с повышением за последние 5 лет
● Средняя степень удовлетворенности
● Среднее количество проектов

### 8.1.1. Доля сотрудников с повышением за последние 5 лет среди уволившихся

In [15]:
hr_left = hr[hr.left == 1]
hr_left.head()

satisfaction_level  last_evaluation  number_project  average_montly_hours  \
0                0.38             0.53               2                   157   
1                0.80             0.86               5                   262   
2                0.11             0.88               7                   272   
3                0.72             0.87               5                   223   
4                0.37             0.52               2                   159   

   time_spend_company  Work_accident  left  promotion_last_5years department  \
0                   3              0     1                      0      sales   
1                   6              0     1                      0      sales   
2                   4              0     1                      0      sales   
3                   5              0     1                      0      sales   
4                   3              0     1                      0      sales   

   salary  
0     low  
1  medium  
2  medium  
3     low  
4     low

In [16]:
hr_left['promotion_last_5years'].count()

3571

In [17]:
hr_left['promotion_last_5years'].value_counts()

0    3552
1      19
Name: promotion_last_5years, dtype: int64

In [18]:
(hr_left['promotion_last_5years'] == 1).sum()

19

In [19]:
(hr_left['promotion_last_5years'] == 1).sum() * 100 / hr_left['promotion_last_5years'].count()

0.5320638476617194

### 8.2.1. Доля сотрудников с повышением за последние 5 лет среди оставшихся в компании

In [20]:
hr_stay = hr[hr.left == 0]
hr_stay.head()

satisfaction_level  last_evaluation  number_project  \
2000                0.58             0.74               4   
2001                0.82             0.67               2   
2002                0.45             0.69               5   
2003                0.78             0.82               5   
2004                0.49             0.60               3   

      average_montly_hours  time_spend_company  Work_accident  left  \
2000                   215                   3              0     0   
2001                   202                   3              0     0   
2002                   193                   3              0     0   
2003                   247                   3              0     0   
2004                   214                   2              0     0   

      promotion_last_5years department salary  
2000                      0      sales    low  
2001                      0      sales    low  
2002                      0      sales    low  
2003                      0      sales    low  
2004                      0      sales    low

In [21]:
hr_stay['promotion_last_5years'].count()

11428

In [22]:
hr_stay['promotion_last_5years'].value_counts()

0    11128
1      300
Name: promotion_last_5years, dtype: int64

In [23]:
(hr_stay['promotion_last_5years'] == 1).sum()

300

In [24]:
(hr_stay['promotion_last_5years'] == 1).sum() * 100 / hr_stay['promotion_last_5years'].count()

2.625131256562828

#### Вывод: по данным анализа доля сотрудников с повышениями за последние 5 лет больше среди оставшихся в компании (300 человек или 2.6% по отношению к 19 сотрудникам или 0.53% соответственно)

### 8.1.2. Средняя степень удовлетворенности среди уволившихся

In [27]:
hr_left['satisfaction_level'].mean()

0.44009801176140917

### 8.2.2. Средняя степень удовлетворенности среди оставшихся в компании сотрудников

In [26]:
hr_stay['satisfaction_level'].mean()

0.666809590479516

#### Вывод: средняя степень удовлетворенности среди сотрудников, оставшихся в компании выше

### 8.1.3. Среднее количество проектов среди уволившихся

In [19]:
hr_left['number_project'].mean()

3.8555026603192384

### 8.2.3. Среднее количество проектов среди оставшихся в компании сотрудников

In [28]:
hr_stay['number_project'].mean()

3.786664333216661

#### Вывод: среднее количество проектов среди уволившихся сотрудников незначительно выше